rockcraft/services/package.py

```diff
@@ -163,7 +163,13 @@ def _pack(
 
     emit.progress("Adding Pebble entrypoint")
 
-    new_image.set_entrypoint()
+    if project.services and project.entrypoint_service in project.services:
+        new_image.set_entrypoint(
+            project.entrypoint_service,
+            project.services[project.entrypoint_service].command,
+        )
+    else:
+        new_image.set_entrypoint(None, None)
```

Comment:
I think this should be split. It's better to have pure functions that only do one job. It also makes it easier for unit testing. So, I propose:

 - 2 functions: `new_image.set_entrypoint(entrypoint_service)` and `new_image.set_cmd(entrypoint_service_command)` 
 - then this code block becomes:
```python
# you let the function handle the `None` value
new_image.set_entrypoint(project.entrypoint_service)

if project.entrypoint_service:
    # you let this new function handle the splitting of the command
    new_image.set_cmd(project.services[project.entrypoint_service].command)
```